ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java

```diff
@@ -1106,6 +1118,64 @@ private void setInstanceConfig(String dcName, String instanceName, String resour
   }
 
+  /**
+   * Set resource configs for waged rebalancer
+   * @param dcName data center
+   * @param resourceName resource
+   * @param configAccessor the {@link ConfigAccessor} to access configuration in Helix.
+   * @param propertyStoreToDataNodeConfigAdapter {@link PropertyStoreToDataNodeConfigAdapter} to access property store data.
+   * @param helixAdmin {@link HelixAdmin}
+   * @param clusterConfigFields inputted cluster configuration
+   */
+  private void setResourceConfig(String dcName, String resourceName, ConfigAccessor configAccessor,
+      PropertyStoreToDataNodeConfigAdapter propertyStoreToDataNodeConfigAdapter, HelixAdmin helixAdmin,
+      ClusterConfigFields clusterConfigFields) throws IOException {
+
+    ResourceConfig resourceConfig = configAccessor.getResourceConfig(clusterName, resourceName);
+    if (resourceConfig == null) {
+      ResourceConfig.Builder resourceBuilder = new ResourceConfig.Builder(resourceName);
+      resourceConfig = resourceBuilder.build();
+    }
+
+    // Get weights for Partitions of the resource from property store
+    Map<String, Map<String, Integer>> partitionCapacityMap = new HashMap<>();
+
+    // 1. Get list of instances hosting partitions of this resource
+    IdealState idealState = helixAdmin.getResourceIdealState(clusterName, resourceName);
+    Set<String> partitions = idealState.getPartitionSet();
+    Set<String> instances = new HashSet<>();
+    for (String partition : partitions) {
+      instances.addAll(idealState.getInstanceSet(partition));
+    }
+
+    // 2. Get weights of partitions residing on these instances from property store
+    for (String instanceName : instances) {
+      DataNodeConfig dataNodeConfig =
+          getDataNodeConfigFromHelix(dcName, instanceName, propertyStoreToDataNodeConfigAdapter, null);
+      // Get Disk configs on the node
+      Map<String, DataNodeConfig.DiskConfig> diskConfigs = dataNodeConfig.getDiskConfigs();
+      for (DataNodeConfig.DiskConfig diskConfig : diskConfigs.values()) {
+        Map<String, DataNodeConfig.ReplicaConfig> replicaConfigs = diskConfig.getReplicaConfigs();
+        // Get replica configs for each replica on the disk
+        for (Map.Entry<String, DataNodeConfig.ReplicaConfig> replicaConfigEntry : replicaConfigs.entrySet()) {
+          String partitionName = replicaConfigEntry.getKey();
+          long capacityInGB = replicaConfigEntry.getValue().getReplicaCapacityInBytes() / 1024 / 1024 / 1024;
+          partitionCapacityMap.put(partitionName, new HashMap<>());
+          partitionCapacityMap.get(partitionName)
+              .put(DISK_KEY, Math.toIntExact(capacityInGB) + PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB);
+        }
+      }
+    }
+    partitionCapacityMap.put(DEFAULT_PARTITION_KEY, new HashMap<>());
+    partitionCapacityMap.get(DEFAULT_PARTITION_KEY).put(DISK_KEY, clusterConfigFields.partitionDiskWeightInGB);
```

Comment:
should add PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB.

Reply:
Thank you for catching! Done